test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -239,8 +249,23 @@ struct test_transform_iterator {
         test_transform_effect(in1.begin(),  in1.end(),  in2.begin());
         test_transform_effect(in1.cbegin(), in1.cend(), in2.begin());
 
-        auto new_transform_iterator = oneapi::dpl::make_transform_iterator(in2.begin(), [](T2& x) { return x + 1; });
-        test_random_iterator(new_transform_iterator);
+        transform_functor new_functor;
+        oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it1(in1.begin());
+        oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it2(in1.begin(), new_functor);
+
+        ::std::forward_list<int> f_list{1,2,3,4,5,6};
+        oneapi::dpl::transform_iterator list_it1(f_list.begin(),new_functor);
+        oneapi::dpl::transform_iterator list_it2(f_list.end(),new_functor);
```

Comment:
I believe it doesn't work with C++ versions less than 17

Reply:
Fixed